### ISSUE TITLE
Calendar ARIA issues

### ISSUE THREAD

**Issue by lukereative** [primefaces/primevue#4900]
### Describe the bug

A couple of issues with the implementation of ARIA attributes on the Calendar component:

1. When using the `inline` option on the calendar, the `.p-datepicker` div does not have a role (without the `inline` attribute it has `role="dialog"`). This means that the `aria-label` which by default has the value 'Choose Date' is invalid, since div's aren't an interactive element.

2. The day number 'buttons' which are spans have `aria-selected` on them, which is invalid

### Reproducer

https://codesandbox.io/p/devbox/frosty-pond-98778j

### PrimeVue version

3.42.0

### Vue version

3.x

### Language

TypeScript

### Build / Runtime

Vue CLI App

**Comment by G-Rath** [comment]
Note that it also looks like the component is inconsistent at least with `aria-selected`: it's applied to days, but not to months or years.

I was about to create a new issue about adding it but then came across this which points out its actually invalid to have so I guess really it should be removed when the `view` is set to `day`.

In terms of what to use, it looks like maybe [`aria-current`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current) is correct here?
